docs/devel/networking.md

```diff
@@ -1,3 +1,84 @@
+
+It is important to refer that while this step is performed by virt-handler, it
+is performed in the *target* virt-launcher's net namespace, as can be seen in
+the implementation of the [setPodNetworkPhase1 function](https://github.com/kubevirt/kubevirt/blob/8e7f2a411f5d06f1734e96c7bc08d0bb9ec1d500/pkg/virt-handler/vm.go#L340).
+
+This will in turn call the
+[PlugPhase1](https://github.com/kubevirt/kubevirt/blob/5995d2164faefdcc7a59a857e12f8ed4cdb7e094/pkg/virt-launcher/virtwrap/network/podinterface.go#L84),
+which performs the following operations:
+- `getPhase1Binding`: this function initializes the interface structures, and
+  selects the correct `driver` to use. The driver, is essentially a
+  `BindMechanism`. The next operations will all be of specific implementations
+  of [BindMechanisms](#binding-mechanisms).
+
+- `discoverPodNetworkInterface`: Each `BindMechanism` requires different
+   information about the pod interface - slirp, for instance, doesn't require
+   any info. The others, gather the following information:
+   - IP address
+   - Routes (**only** bridge)
+   - Gateway
+   - MAC address (**only** bridge)
+   - Link MTU
+
+- `preparePodNetworkInterfaces`: this function will make use of the
+  aforementioned information, performing actions specific to each `BindMechanism`.
+  See each `BindMechanism` section for more details.
+
+- `setCachedInterface`: caches the interface in memory
+
+- `setCachedVIF`: this will persist the
+  [VIF](https://github.com/kubevirt/kubevirt/blob/51ce9d8e6def33c0c260f0cdb994f08ae5b9938e/pkg/virt-launcher/virtwrap/network/common.go#L48)
+  object in the file system, making the configured pod interface available to
+  the virt-launcher pod. The VIF is cached in
+  `/proc/<virt-launcher-pid>/root/var/run/kubevirt-private/vif-cache-<iface_name>.json`.
+
+### Unprivileged VMI networking configuration
+The virt-launcher is an untrusted component of KubeVirt (since it wraps the
+libvirt process that will run third party workloads). As a result, it must be
+run with as little privileges as required. As of now, the only capability
+required by virt-launcher to configure networking is the `CAP_NET_ADMIN`
+capability.
+
+In this second phase, virt-launcher retrieves the configuration data previously
+gathered in phase #1 (loads the cached VIF object), and decorates the domain
+xml configuration of the VM it will encapsulate. This *decoration* is specific
+to each [binding mechanism](#binding-mechanisms), as each of those can involve
+different libvirt configurations.
+
+## Binding Mechanisms
+A binding mechanism can be seen as the translation service between KubeVirt's
+API and Libvirt's domain xml. Each interface type has a different binding
+mechanism, since it will lead to a different libvirt domain xml specification.
+
+As of now, the existent binding mechanisms are:
+- [bridge](#bridge-binding-mechanism)
+- [masquerade](#masquerade-binding-mechanism)
+- [slirp](#slirp-binding-mechanism)
```

Comment:
You missed SR-IOV

Reply:
SR-IOV is not a [BindMechanism](https://github.com/kubevirt/kubevirt/blob/3860036877d3f39dc891b90867783abb421e1b93/pkg/virt-launcher/virtwrap/network/podinterface.go#L46). 

It is an [InterfaceBindingMethod](https://github.com/kubevirt/kubevirt/blob/e66549fff089a947c7a8763ea1645cdbb60b7950/staging/src/kubevirt.io/client-go/api/v1/schema.go#L1015), which I never mentioned throughout the doc.

Do you think that introducing that concept is necessary ?

Reply:
Well at least I would learn the difference. It is not clear to me what is the difference, but I will not fight for the explanation on this PR.

Reply:
I think if needed, it would definitely belong on a separate PR; the scope of this PR is to explain the 2 phases of pod networking configuration & the bind mechanism concept. 

The difference is simple though; binding mechanism are actors in the 2 phases, follow the aforementioned interface, and require interface creation / configuration. 

An `InterfaceBindingMethod` is a simple element of the kubevirt API; it is used only in the [converter](https://github.com/kubevirt/kubevirt/blob/00a9c07f4a065de0ce4d99058f159b8cf4cbb456/pkg/virt-launcher/virtwrap/api/converter.go#L1150) - phase#2. No device needs to be created or configured - there's just conversion from kubevirt API to dom xml.

This comment can work as a teaser until I add that section :)

Reply:
There are 3 phases! 1. CNI and 2.+3. binding of VM to the pod network :P 

 I see, thanks for the explanation!

Reply:
I'll clarify it, thanks for keeping me honest.

Reply:
This difference is unfortunate. The abstraction should take care of the "how", to the outside (including the code layers) it should be the same... binding a bridge or an SRIOV device should look the same in the API.
But this is another topic, not related to expressing what exists.